tidy3d/components/simulation.py

```diff
@@ -2787,3 +2790,142 @@ def allow_gain(self) -> bool:
             elif medium.allow_gain:
                 return True
         return False
+
+    @staticmethod
+    def _sel_data_inside_bounds(data: SpatialDataArray, bounds: Bound) -> Tuple[bool, SpatialDataArray]:
```

Comment:
a few things that might help make this more 

1. Maybe this should be a DataArray method? to avoid Simulation getting even larger?
2. does it make sense to separate `is_full_cover` and `sel_data_inside_bounds` into two functions? for example `sel_data_inside_bounds` does the data selection and `is_full_cover` just looks at bounds compared to self's coordinates?

Reply:
imlpemented